docs/reference/contributing/target/QuadSPI.md

```diff
@@ -1,25 +1,50 @@
 <h3 id="quadspi-port">QuadSPI</h3>
 
-[Include a brief description here.]
+The QSPI HAL defines API for targets that contain QSPI capable peripheral. Developers often use the QSPI interface for data storage.
+
+<span class="warnings">**Warning:** We are changing the QSPI HAL API in an upcoming release of Mbed OS. You can find details on how it may affect you in the [Implementing the QSPI API](#implementing-the-qspi-api) section.
 
 #### Assumptions
 
 ##### Defined behavior
 
-[Include any defined behavior in bullet format here.]
+- A target implementaion covers most of the QSPI frame format (some targets might not provide the flexibility for setting all frame parameters).
+- Command transfer - A target might provide additional API for sending device-specific commands. If it does not, you can implement it using read and write functions. (This is target or driver dependent.)
 
 ##### Undefined behavior
 
-[Include any undefined behavior in bullet format here.]
+- Calling any function other than `qspi_init` before the initialization of the QSPI.
 
```

Comment:
Please add a "Notes" section with any common trouble areas developers may come across when implementing this API (if applicable).

Reply:
Not yet, will be filled later once we get more targets and questions about the porting